### ISSUE TITLE
Remove deprecated arguments for 1.0 release

### ISSUE THREAD

**Issue by jgabry** [stan-dev/cmdstanr#409]
Just opening this so I can add it to the v1.0.0 milestone. There are many deprecated arguments we should remove in v1.0.0. (Alternatively, since we're still in beta we could just remove them in a future beta release since we've already done several beta releases keeping the deprecated arguments.)

**Comment by jgabry** [comment]
Once #520 is merged we'll also need to remove support for using old .cmdstanr directory instead of .cmdstan. Removing this would entail removing the `old` argument to `cmdstan_default_path` and `cmdstan_default_install_path` and removing the use of `cmdstan_default_path(old = TRUE)` from `zzz.R`.
